### PR TITLE
frontend-fix-2309-bug-fixes YS bugs fixes

### PR DESCRIPTION
**Pull request by S-PRAVINKUMAR**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Bugs fixed : 16, 18, 7, 26, 10, 27

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/Controller.purs [hunk 1]
```diff
@@ -160,11 +160,18 @@ eval (CallBackImageUpload image imageName imagePath) state = if(state.props.clic
...
-    _ -> continue state
+eval (DatePicker (label) resp year month date) state = do
+  case resp of 
+    "SELECTED" -> case label of
+                    "DATE_OF_BIRTH" -> continue state {data = state.data { dob = (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00" , dobView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year)}
+                                                        , props {isDateClickable = true }}
+                    "DATE_OF_ISSUE" -> continue state {data = state.data { dateOfIssue = Just $ (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00" , dateOfIssueView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year), imageFront = "null"}
+                                                        , props {isDateClickable = true }} -- imageFront made null to handle fallback
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/Controller.purs:169]: remove commented line

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/View.purs [hunk 2]
```diff
@@ -203,7 +203,7 @@ reEnterLicenceNumber state push =
       [ width MATCH_PARENT
       , height WRAP_CONTENT
       , text (getString SAME_REENTERED_DL_MESSAGE)
-      , visibility $ if (DS.toLower(state.data.driver_license_number) /= DS.toLower(state.data.reEnterDriverLicenseNumber)) then VISIBLE else GONE
+      , visibility $ if (DS.toLower(state.data.driver_license_number) /= DS.toLower(state.data.reEnterDriverLicenseNumber) && (DS.length state.data.reEnterDriverLicenseNumber /= 0)) then VISIBLE else GONE
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/View.purs:206]: you can use `DS.null` for strings

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/Controller.purs [hunk 3]
```diff
@@ -160,11 +160,18 @@ eval (CallBackImageUpload image imageName imagePath) state = if(state.props.clic
...
-    "DATE_OF_ISSUE" -> continue state {data = state.data { dateOfIssue = Just $ (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00" , dateOfIssueView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year), imageFront = "null"}} -- imageFront made null to handle fallback
-    _ -> continue state
+eval (DatePicker (label) resp year month date) state = do
+  case resp of 
+    "SELECTED" -> case label of
+                    "DATE_OF_BIRTH" -> continue state {data = state.data { dob = (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00" , dobView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year)}
+                                                        , props {isDateClickable = true }}
+                    "DATE_OF_ISSUE" -> continue state {data = state.data { dateOfIssue = Just $ (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00" , dateOfIssueView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year), imageFront = "null"}
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/Controller.purs:168]: you can store date in variable and can use same in both the cases like 
let date = (dateFormat year) <> "-" <> (dateFormat (month+1)) <> "-" <> (dateFormat date) <> " 00:00:00.233691+00"
    dateView = (show date) <> "/" <> (show (month+1)) <> "/" <> (show year)

File: Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/ComponentConfig.purs [hunk 4]
```diff
@@ -89,7 +91,7 @@ primaryEditTextConfigReEnterDl state = let
           , capsLock = true
           , color = Color.black800
         }
-      , stroke = if (DS.toLower(state.data.driver_license_number) /= DS.toLower(state.data.reEnterDriverLicenseNumber)) then ("1," <> Color.red) else ("1," <> Color.borderColorLight)
+      , stroke = if (DS.toLower(state.data.driver_license_number) /= DS.toLower(state.data.reEnterDriverLicenseNumber) && (DS.length state.data.reEnterDriverLicenseNumber /= 0)) then ("1," <> Color.red) else ("1," <> Color.borderColorLight)
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/UploadDrivingLicenseScreen/ComponentConfig.purs:94]: use `DS.null`

File: Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs [hunk 5]
```diff
@@ -351,8 +351,8 @@ vehicleRegistrationNumber state push =
                 [ width MATCH_PARENT
                 , height WRAP_CONTENT
                 , text (getString SAME_REENTERED_RC_MESSAGE)
-                , visibility $ if (DS.toLower(state.data.vehicle_registration_number) /= DS.toLower(state.data.reEnterVehicleRegistrationNumber)) then VISIBLE else GONE
-                , color Color.darkGrey
+                , visibility $ if (DS.toLower(state.data.vehicle_registration_number) /= DS.toLower(state.data.reEnterVehicleRegistrationNumber) && (DS.length state.data.reEnterVehicleRegistrationNumber) /= 0 ) then VISIBLE else GONE
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/AddVehicleDetailsScreen/View.purs:354]: use `DS.null`

File: Frontend/ui-common/src/Engineering/Helpers/Utils.purs [hunk 6]
```diff
@@ -56,4 +68,22 @@ showAndHideLoader delayInMs title description state = do
...
+    case config' of
+      Just config -> do
+        case runExcept (decode (encode config )) of
+            Right (obj :: AppConfig) -> config
+            Left err ->DefaultConfig.config
+      Nothing -> do
+            DefaultConfig.config
+
+getConfig :: forall  a. Effect (Maybe a)
+getConfig = getMerchantConfig Just Nothing
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/Utils.purs:89]: can we not pass `Just` and `Nothing` to JS to get the response in `Maybe` type instead we can have two function one which gets response and one which will add wrapper based on response

File: Frontend/ui-common/src/Engineering/Helpers/Utils.purs [hunk 7]
```diff
@@ -56,4 +68,22 @@ showAndHideLoader delayInMs title description state = do
...
     pure unit
-  pure unit
\ No newline at end of file
+  pure unit
+
+getAppConfig :: FlowBT String AppConfig
+getAppConfig = liftFlowBT $ getAppConfig_
+
+getAppConfig_ :: Effect AppConfig
+getAppConfig_  = do
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/Utils.purs:77]: change function name to `getAppConfigImpl` or rename to something else

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js [hunk 8]
```diff
@@ -21,11 +21,12 @@ window.getMerchantConfig = function () {
     "BONUS_EARNED" : "true",
     "clientName" : "Namma Yatri",
     "languageList" : [
-      {"name":"English","value":"EN_US", "subtitle": ""}, 
+      {"name":"English","value":"EN_US", "subtitle": "ಆಂಗ್ಲ"},
```
[the remark this answers is not in the document]
Comment (S-PRAVINKUMAR) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js:24]: Yes. 7th issue

[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js:24]: okay

[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js:24]: than why others subtitle is in english ?

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (S-PRAVINKUMAR) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/v1-configuration.js:24]: Yes. I also noticed. but Akash only said, this the way you want to do.

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**